File: LeetCode/122. Best Time to Buy and Sell Stock II/solution.cs
public class Solution {
    public int MaxProfit(int[] prices) {

        int buyPrice = prices[0];
        int cumulativeProfit = 0;

        for(int i = 1; i < prices.Length; i++){
            if(buyPrice < prices[i]){
                cumulativeProfit += prices[i] - buyPrice;
            }

            buyPrice = prices[i];
        }


        return cumulativeProfit;
    }
}

File: LeetCode/122. Best Time to Buy and Sell Stock II/solution.cpp
#include <vector>
using namespace std;


/*
Runtime 3ms Beats 88.15%
Memory 15.37MB Beats 92.62%
*/
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buy = __INT_MAX__;
        int profit = 0;

        for(int price : prices){
            if(price > buy)
                profit += price - buy;
            
            buy = price;
        }

        return profit;
    }
};